pkg/controller/traffic/utils_test.go

```diff
@@ -0,0 +1,104 @@
+package traffic
+
+import (
+	"fmt"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	shipperfake "github.com/bookingcom/shipper/pkg/client/clientset/versioned/fake"
+	shipperinformers "github.com/bookingcom/shipper/pkg/client/informers/externalversions"
+	shippertesting "github.com/bookingcom/shipper/pkg/testing"
+	corev1 "k8s.io/api/core/v1"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/record"
+)
+
+// NOTE: this does not try to implement endpoint subsets at all. All
+// pods are assumed to be part of the same subset and to be always ready.
+func shiftPodInEndpoints(pod *corev1.Pod, endpoints *corev1.Endpoints) *corev1.Endpoints {
+	var podGetsTraffic bool
+	trafficStatusLabel, ok := pod.Labels[shipper.PodTrafficStatusLabel]
+	if ok {
+		podGetsTraffic = trafficStatusLabel == shipper.Enabled
+	}
+
+	addresses := endpoints.Subsets[0].Addresses
```

Comment:
Are addresses sorted? If yes, a suggested edit is to use `sort.Search()`. If no, please, never mind :-)

Reply:
Since this is a test environment, we control the object entirely. So I guess not, not guaranteed to be sorted.